Docs release 1.9.0 csi 3911 node get volume stats

rivkap opened the pull request.

oriyarde commented on docs/content/release_notes/whats_new.md:
```diff
@@ -8,6 +8,20 @@ IBM® block storage CSI driver 1.9.0 introduces the enhancements that are detail
+- kubelet_volume_stats_inodes_used
+- kubelet_volume_stats_used_bytes
+
+For more information on using kubelet volume metrics, see the [Container Storage Interface (CSI) spec documentation for `NodeGetVolumeStats`](https://github.com/container-storage-interface/spec/blob/v1.5.0/spec.md#nodegetvolumestats).
```
```suggestion
For more information about each metric, see `VolumeUsage` in the [Container Storage Interface (CSI) spec documentation for `NodeGetVolumeStats`](https://github.com/container-storage-interface/spec/blob/v1.5.0/spec.md#nodegetvolumestats).
```

rivkap replied:
updated

oriyarde reviewed (commented):
IMO the base branch for this PR should either be develop, or a general 1.9.0 docs branch (and not "stretched feature" branch)

rivkap commented on docs/content/release_notes/whats_new.md:
```diff
@@ -8,6 +8,20 @@ IBM® block storage CSI driver 1.9.0 introduces the enhancements that are detail
+## New metrics support
+
+IBM® block storage CSI driver 1.9.0 introduces new kubelet volume metric support. Use 
+`NodeGetVolumeStats` to retrieve volume metrics. The following metrics are currently supported:
```
Is this more correct?

"Use the `NodeGetVolumeStats` Remote Procedure Call (RPC) call to retrieve the volume metrics with a metrics scraper."

oriyarde replied:
no, the verb "use" addresses the user, but a user cannot directly issue remote procedure calls.
possible metrics info flows:
`csi node` --gRPC--> `kubelet` --HTTP--> `user`
`csi node` --gRPC--> `kubelet` --HTTP--> `prometheus` --?--> `user`
invalid flow:
`csi node` --gRPC--> `user`
@matancarmeli7 is the feature owner, he can elaborate

rivkap replied:
@matancarmeli7  / @oriyarde  Please help provide an appropriate sentence that I can work with.

Maybe more like - 
"The `NodeGetVolumeStats` Remote Procedure Call (RPC) call enables retrieval of the volume metrics with a metrics scraper."?

or maybe we don't need this sentence all together? Do we need to mention NodeGetVolumeStats?

oriyarde replied:
that is correct, but as your questions suggest, I agree we don't need to mention `NodeGetVolumeStats`, that is an implementation detail

rivkap replied:
is the link that we provide enough information for implementation?

oriyarde replied:
yes, though I think the link I wrote [here](https://github.com/IBM/ibm-block-csi-driver/pull/460#discussion_r817495108) could be useful for how to use this feature

matancarmeli7 replied:
IMO we don't need to mention `NodeGetVolumeStats`, we need to say something like this: "We are supporting exposing various metrics through `kubelet` on mounted volumes that have been created by IBM block CSI" (the english can be better, but this is the direction I think is good) and add the relevant metrics, link that Ori sent [here](https://github.com/IBM/ibm-block-csi-driver/pull/460#discussion_r817495108) and the spec link that we already provided. 
What do you think?

rivkap replied:
updated - please approve

rivkap commented:
> IMO the base branch for this PR should either be develop, or a general 1.9.0 docs branch (and not "stretched feature" branch)
I based it off of stretched so it would only show the changes made for this specific feature.
i'll end up merging it into my main docs branch once i know the technical aspects are approved

oriyarde commented:
> > IMO the base branch for this PR should either be develop, or a general 1.9.0 docs branch (and not "stretched feature" branch)
> > I based it off of stretched so it would only show the changes made for this specific feature.
> > i'll end up merging it into my main docs branch once i know the technical aspects are approved

it actually shows [irrelevant diff now](https://github.com/IBM/ibm-block-csi-driver/pull/460/files#diff-aa2fcc72291bf5690f95700a4df50703346a86fe5e05b50d438f4e7e94917521), and has conflicts

rivkap commented:
> > > IMO the base branch for this PR should either be develop, or a general 1.9.0 docs branch (and not "stretched feature" branch)
> > > I based it off of stretched so it would only show the changes made for this specific feature.
> > > i'll end up merging it into my main docs branch once i know the technical aspects are approved
> 
> it actually shows [irrelevant diff now](https://github.com/IBM/ibm-block-csi-driver/pull/460/files#diff-aa2fcc72291bf5690f95700a4df50703346a86fe5e05b50d438f4e7e94917521), and has conflicts

I changed the branch to merge back into when I merged the stretched branch.
the **2** files that I changed are the **what's new file** and the **limitations** files. everything else is irrelevant for this specific branch. I'll resolve any conflicts before i merge into my main branch.
please provide reviews and approvals based off of these two files. thanks

rivkap commented:
opening with a different PR
